index.js

```diff
@@ -94,7 +103,24 @@ function resultPrinter(options, source, removedNodes) {
         var node = removedNodes[i];
         result += source.slice(lastPos, node.start);
         lastPos = node.end;
-        if (!pretty) {
+        if (commentTypes) {
+          // Remove nested comments with a regexp replace
+          var toComment = source.slice(node.start, node.end).replace(NESTED_COMMENTS_RX, ' ').replace(/\s+;/, ';');
+          if (!node.loc || node.loc.start.line === node.loc.end.line) {
+            // possibly use the shorter single ':' syntax
+            if (toComment && toComment[0] === ':') {
+              result += ' /*' + toComment + ' */ ';
+            } else {
+              result += '/*:: ' + toComment + ' */';
+            }
+          } else {
+            var toCommentLines = toComment.split(LINE_RX);
+            // TODO: detect file line endings scheme (\n or \r\n)
```

Comment:
We should probably resolve this TODO. I wouldn't want to merge this feature if it messes up Windows line endings.